File: 백준/Silver/1065. 한수/한수.py
N = int(input())

cnt = 0
for i in range(1, N+1):
    if 1 <= i <= 9:
        cnt += 1
        continue

    d = int(str(i)[1])-int(str(i)[0])
    check = True
    for j in range(len(str(i))-1):
        if int(str(i)[j+1])-int(str(i)[j]) != d:
            check = False
            break
    if check:
        cnt += 1

print(cnt)

File: 백준/Silver/1065. 한수/한수.cc
#include <iostream>
using namespace std;

int main() {
    int cnt = 0;

    int N;
    scanf("%d", &N);
    for (int i=1; i<=N; i++) {
        if (i < 100) cnt++;
        else if (i/100 - (i%100)/10 == (i%100)/10 - (i%100)%10) cnt++;
    }
    printf("%d", cnt);
    return 0;
}